test.py

```diff
@@ -169,7 +169,7 @@ def validate_nwbs():
                 is_family_nwb_file = False
                 try:
                     with pynwb.NWBHDF5IO(nwb, mode='r') as io:
-                        errors = pynwb.validate(io)
+                        errors = validate(io, use_cached_namespaces=False)  # previously io did not validate against cached namespaces
```

Comment:
```suggestion
                        errors = validate(io, use_cached_namespaces=False),
                        errors.append(validate(io, use_cached_namespaces=True))
```
1. I think the original comment here might not make sense in the future
2. The example NWB files are all generated using the version of NWB being tested, and because pynwb caches the spec by default, there should be no difference between using cached namespaces and not. 
3. Since the `pynwb.validate` and `pynwb-validate` should be the same now, we don't really need this test anymore since we have the `pynwb-validate` test below. But since this `validate_nwbs()` function is super conservative in its testing of every combination, then for consistency, I suggest we validate with both `use_cached_namespaces=True` and `use_cached_namespaces=False`.

Reply:
I think these lines: https://github.com/NeurodataWithoutBorders/pynwb/blob/80fc386c20d791208142f97e8cfacbce967b8aee/docs/gallery/general/extensions.py#L41 

https://github.com/NeurodataWithoutBorders/pynwb/blob/80fc386c20d791208142f97e8cfacbce967b8aee/docs/gallery/general/extensions.py#L267

could be updated to:
`ns_builder.include_namespace("core")`

to address this issue.

Reply:
Interesting. For now, let's update that to `ns_builder.include_namespace("core")`.

We need to think through what it means to run the pynwb validator on a file, validating against a particular non-core namespace. Should the core namespace always be included during validation, regardless of whether the extension includes the core namespace, and the choices are either use the core namespace cached in the file or the one installed with pynwb? I think so...

I think we eventually want to move toward not even allowing validation against a particular namespace. Either the file is valid given its cached namespaces (or the namespaces installed by pynwb and loaded by the user), or the file is not. Otherwise, we run into weird issues with such as:
https://github.com/hdmf-dev/hdmf/issues/608 and https://github.com/hdmf-dev/hdmf/issues/525

I would say let's make the above change for now, and iterate on these other ideas in a separate PR which does not need to make it in pynwb 3.0.0rc1.

Reply:
> Should the core namespace always be included during validation, regardless of whether the extension includes the core namespace, and the choices are either use the core namespace cached in the file or the one installed with pynwb? I think so...

I think so too?  But agreed it would definitely be helpful to compile and discuss all these different scenarios to get a better idea of the end goal validation behavior

> I would say let's make the above change for now, and iterate on these other ideas in a separate PR which does not need to make it in pynwb 3.0.0rc1

Sounds good!